server/core/src/main/java/io/whitefox/core/services/AbstractSharedTable.java

```diff
@@ -0,0 +1,15 @@
+package io.whitefox.core.services;
+
+import io.whitefox.core.Metadata;
+import io.whitefox.core.ReadTableRequest;
+import io.whitefox.core.ReadTableResultToBeSigned;
+import java.util.Optional;
+
+public interface AbstractSharedTable {
```

Comment:
What about `SharedTable` as a name? I think fits more with the conventions used until now in the project

Reply:
I think we should think about names a bit more :) Having two different things with the same possible name is a bad idea...

Reply:
Please open a followup issue on naming/renaming, quoting this homonymy case